src/List/List.js

```diff
@@ -45,13 +49,16 @@ export default class List extends Component {
       className: classNameProp,
       component,
       padding,
+      children,
       ...other,
     } = this.props;
     const classes = this.context.styleManager.render(styleSheet);
+    const firstChild = Children.toArray(children)[0];
     const className = classNames(classes.root, {
       [classes.padding]: padding,
+      [classes.subheader]: isValidElement(firstChild) && firstChild.type === Subheader,
```

Comment:
Regarding
```js
firstChild.type === Subheader
```
Some time ago, we had a discussion on the best approach. The conclusion was to use a `muiName` string in order to:
1. Remove the dependency injection
2. Allow users to mess around
I believe that we should follow this approach here too.